packages/web-components/src/player-component/player.class.ts

```diff
@@ -668,14 +668,15 @@ export class PlayerWrapper {
         const MAX_LATENCY_WINDOW = 3; // in seconds
         this._driftCorrectionTimer = window.setInterval(async () => {
             const video = this.player.getMediaElement() as HTMLMediaElement;
-            if (this.player.getBufferedInfo().video[0].end - MAX_LATENCY_WINDOW > video.currentTime &&
-                !video.paused && this.isLive && !this.player.isBuffering()
+            const buffer = this.player.getBufferedInfo().video[0];
```

Comment:
Currently, shaka-player won't return null/undefined from getBufferedInfo(), but the video member can definitely be the empty array. You must therefore check whether video.length > 0. I would actually put this and the check for it INSIDE your if-statement. You should also check for video null. In other words:

if (video && !video.paused && this.isLive && !this.player.isBuffering()) {
    const videoBuffered = this.player.getBufferedInfo().video;
    const videoCurrentTime = video.currentTime; // Lock in the current value
    const videoBufferedEnd = videoBuffered.length > 0 ? videoBuffered[videoBuffered.length - 1].end : videoCurrentTime - 1;
    if (videoBufferedEnd - MAX_LATENCY_WINDOW > videoCurrentTime) {
        [... be sure to replace video.currentTime with videoCurrentTime...]
    }
}

Reply:
Actually I'd put const videoCurrentTime above const videoBuffered so that const VideoBufferedEnd is right after const videoBuffered.